frontend/src/Commons/Forms/InputModules/Calculation/Calculation.tsx

```diff
@@ -140,134 +145,146 @@ const CalculationComponent: React.FC<InputModuleProps> = ({
+                  onChange={(e) => handleFieldChange(e, setFieldValue)}
                 />
-                <StyledSmallFormGroup>
-                  <CustomField
-                    label="Application Rate"
-                    name="applRate"
-                    id="applRate"
-                    type="number"
-                    width="50%"
-                  />
+              </StyledFieldSelect>
+              <StyledDivider />
+              <StyledGroupFormView>
+                <StyledLeftView>
                   <CustomSelect
-                    label="Units"
-                    name="applUnitId"
-                    id="applUnitId"
-                    options={
-                      fertilizersDetails[selectedIndex]?.fertilizerTypeId.includes('Dry Fertilizer')
-                        ? DryApplicationUnits
-                        : fertilizersDetails[selectedIndex]?.fertilizerTypeId.includes(
-                              'Liquid Fertilizer',
-                            )
-                          ? LiquidApplicationUnits
-                          : []
-                    }
-                    width="50%"
+                    label="Fertilizer"
+                    id="fertilizerId"
+                    name="fertilizerId"
+                    options={fertilizerOption}
+                    width="100%"
                     formCalc
-                    onChange={(e) => handleChange(e, setFieldValue)}
+                    onChange={(e) => handleFertilizerChange(e, setFieldValue)}
                   />
-                </StyledSmallFormGroup>
-                {fertilizersDetails[selectedIndex]?.fertilizerTypeId.includes('Liquid') && (
                   <StyledSmallFormGroup>
                     <CustomField
-                      label="Density"
-                      name="liquidDensity"
-                      id="liquidDensity"
+                      label="Application Rate"
+                      name="applRate"
+                      id="applRate"
                       type="number"
                       width="50%"
                     />
                     <CustomSelect
-                      label="Density Units"
-                      id="liquidDensityUnitId"
-                      name="liquidDensityUnitId"
-                      options={DensityUnits}
+                      label="Units"
+                      name="applUnitId"
+                      id="applUnitId"
+                      options={
+                        fertilizersDetails[selectedIndex]?.fertilizerTypeId.includes(
+                          'Dry Fertilizer',
+                        )
+                          ? DryApplicationUnits
+                          : fertilizersDetails[selectedIndex]?.fertilizerTypeId.includes(
+                                'Liquid Fertilizer',
+                              )
+                            ? LiquidApplicationUnits
+                            : []
+                      }
```

Comment:
You could move this to a function in the component instead of this nested ternary